packages/cli/src/constructs/multi-step-check.ts

```diff
@@ -0,0 +1,141 @@
+import * as path from 'path'
+import { Check, CheckProps } from './check'
+import { Session } from './project'
+import { Parser } from '../services/check-parser/parser'
+import { CheckConfigDefaults } from '../services/checkly-config-loader'
+import { pathToPosix } from '../services/util'
+import { Content, Entrypoint } from './construct'
+import CheckTypes from '../constants'
+import { CheckDependency } from './browser-check'
+
+export interface MultiStepCheckProps extends CheckProps {
+  /**
+   * A valid piece of Node.js javascript code describing a multi-step interaction
+   * with the Puppeteer or Playwright frameworks.
+   */
+  code: Content|Entrypoint
+  /**
+   * A valid fully qualified domain name (FQDN) to check for SSL certificate
+   * expiration. For example, 'app.checklyhq.com'.
+   */
+  sslCheckDomain?: string
+}
+
+/**
+ * Creates a multi-step Check
+ *
+ * @remarks
+ *
+ * This class make use of the multi-step checks endpoints.
+ */
+export class MultiStepCheck extends Check {
```

Comment:
In the PR description, you mention `Adds multistep check construct, based off of browser construct` but it extends Check and not BrowserCheck. Is it wanted?

Reply:
Yeah it was just based off of the browsercheck construct, meaning they both extend `Check`

Reply:
@ndom91 @Antoine-C we should not add the `slDomain` property

Reply:
Haha we had this discussion elsewhere recently. They're still included in the multistep UI as well. So should we remove them completely from multistep?

Reply:
Do we need also to update the API?

Reply:
Yeah good question, that might be the reason for the failing tests. Doing some more digging here today..